Block/Bmz.php

```diff
@@ -92,13 +92,13 @@ public function _beforeToHtml()
             // Set product data
             $product = $this->registry->registry("product");
             if ($product != null) {
-                $this->addBmzData('id', $product->getId());
+                $this->addBmzData('sku', $product->getId());
             }
 
             // Set category data
-            $category = $this->registry->registry('category');
-            if ($category != null) {
-                $this->addBmzData('categoryid', $category->getId());
+            $categoryObject = $this->registry->registry('current_category');
```

Comment:
Fix for empty categoryid in the BMZs, call to '$this->registry->registry('category')' did not return valid object.